src/system/SystemSocketWatcherLibevent.cpp

```diff
@@ -0,0 +1,115 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      Implements a SystemSocketWatcher using libevent
+ */
+
+#include <support/CodeUtils.h>
+#include <system/SystemLayer.h>
+#include <system/SystemSockets.h>
+
+#include <system/SystemSocketWatcherLibevent.h>
+
+namespace chip {
+namespace System {
+
+namespace {
+
+System::SocketEvents SocketEventFlagsFromLibeventFlags(short eventFlags)
+{
+    return System::SocketEvents()
+        .Set(System::SocketEventFlag::kRead, eventFlags & EV_READ)
+        .Set(System::SocketEventFlag::kWrite, eventFlags & EV_WRITE);
+}
+
+} // anonymous namespace
+
+void SystemSocketWatchState::Init()
+{
+    mEventBase = event_base_new();
+}
+
+void SystemSocketWatchState::HandleEvents()
+{
+    VerifyOrDie(mEventBase != nullptr);
+    event_base_dispatch(mEventBase);
```

Comment:
Should this be event_base_loop?

Reply:
Probably. I've gone back and forth on what exactly `HandleEvents()` should mean, so the second opinion is welcome.